src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -136,6 +137,15 @@ pub struct RawSourceCreationConfig {
     pub params: SourceCreationParams,
 }
 
+impl RawSourceCreationConfig {
+    /// Returns true if this worker is responsible for handling the given partition.
+    pub fn responsible_for<P: Hash>(&self, partition: P) -> bool {
+        let key = usize::cast_from((self.id, partition).hashed());
```

Comment:
This is different now. And the motivation is that partition IDs always start from 0 (or 1?) and we make sure to distribute those differently for different sources?

Reply:
That's right!